notebooks/1_multigenerational_skill.ipynb

```diff
@@ -1,14 +1,5 @@
 {
```

Comment:
<p>Not sure I understand the logic here? Looks like <code>mip_ids</code>  is used without being defined. Maybe you mean to call <code>get_ipcc_dataset()</code>  instead?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/hdrake/cmip6hack-multigen/pull/30/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Reply:
I think that is an old commit. I think the notebooks in the current commit should be self-consistent. These lines have been moved to the `preprocess.py` script and look something like this now:

```python
all_mip_ids = ['far', 'sar', 'tar', 'cmip3', 'cmip5', 'cmip6']

def get_ipcc_collection(mip_ids = all_mip_ids):
    col_dict = {}
    for col_name in mip_ids:
        if 'ar' in col_name:
            json_name = f"https://storage.googleapis.com/ipcc-{col_name}/pangeo-{col_name}.json"
        else:
            json_name = f"https://storage.googleapis.com/cmip6/pangeo-{col_name}.json"
        col = intake.open_esm_datastore(json_name)
        col_dict[col_name] = col
    return col_dict

def load_ensembles(varnames, mip_ids=all_mip_ids, timeslice=None):
    col_dict = get_ipcc_collection(mip_ids=mip_ids)
    ds_dict = load_col_as_dict(col_dict, varnames, timeslice=timeslice)
    ens_dict = model_dict_to_ensemble_dict(ds_dict, varnames)
    return ens_dict
```

Reply:
Ah, I'm sure you're right! But somehow the ReviewNB tool is only showing me that older commit.

Reply:
Ok, I took a "manual" look at the latest commit and I guess I got confused because you don't seem to use `preprocess.py` anywhere in the main notebook `1_multigenerational_skill.ipynb`.

But if you're just asking for my review of the `preprocess.py` script, then I think it looks great. I wouldn't worry too much about a clunky API. We are not trying to write library code here, just trying to make sure our analysis script is reasonably understandable and expressive.

Reply:
That's great, thanks! Yes, I haven't gotten around to updating the `1_multigenerational_skill.ipynb` notebook yet but rather focusing on `0_preprocess_mean_states.ipynb`, which uses the functions in `preprocess.py`.

Reply:
Looks nice and clean.

Reply:
As of https://github.com/hdrake/cmip6hack-multigen/pull/30/commits/0454e4536d1cfbfcef1a4be928ab5552469ab59c, I've now updated the notebook `1_multigenerational_skill.ipynb` which now reads in the < 20 MB worth of data saved from `0_preprocess_mean_states.ipynb` (and stored directly in git for easy access).